### PR TITLE
Doc: HLD of Metadata Backend Document.

### PR DESCRIPTION
**Pull request by VenkyOS**
Adding the HLD of Metadata Backend Document.

### CODE REVIEW

File: doc/HLD_Meta_Data_Back_End.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+- a transaction is a collection of updates. User adds an update to a transaction by capturing the update's region. User explicitly closes a transaction. BE guarantees that a closed transaction is atomic with respect to process crashes that happen after transaction close call returns. That is, after such a crash, either all or none of transaction updates will be present in the segment memory when the segment is opened next time. If a process crashes before a transaction closes, BE guarantees that none of transaction updates will be present in the segment memory;
+
+- a credit is a measure of a group of updates. A credit is a pair (nr, size), where nr is the number of updates and size is total size in bytes of modified regions.
+
+*********************
+Requirements
+*********************
+
+- R.C2.MDSTORE.NUMA: allocator respects NUMA topology
```
Comment (huanghua78) [inline doc/HLD_Meta_Data_Back_End.rst:31]: Make the font BOLD:
```
*R.C2.MDSTORE.NUMA*
```

Comment (huanghua78) [reply doc/HLD_Meta_Data_Back_End.rst:31]: And it's better to rename "C2" to "M0"

Comment (VenkyOS) [reply doc/HLD_Meta_Data_Back_End.rst]: Incorporated.

**Comment by mukul-seagate11** [comment]
@madhavemuri, please review & approved

**Review by mukul-seagate11 — APPROVED** [review]
looks good

File: doc/HLD_Meta_Data_Back_End.rst (hunk 1 above)
Comment (huanghua78) [reply doc/HLD_Meta_Data_Back_End.rst]: I mean, to make all the requirements in this section BOLD.

Comment (VenkyOS) [reply doc/HLD_Meta_Data_Back_End.rst]: Requirements

Comment (VenkyOS) [reply doc/HLD_Meta_Data_Back_End.rst]: The requirements headline is bolded.

Comment (VenkyOS) [reply doc/HLD_Meta_Data_Back_End.rst]: Incorporated.

**Review by huanghua78 — APPROVED** [review]

**Pull request merged**